code/game/objects/items/devices/radio/radio.dm

```diff
@@ -262,8 +262,22 @@ var/global/list/default_medbay_channels = list(
 // I'm not sure who thought that was a good idea. -- Crazylemon
 /mob/living/automatedannouncer
 	var/role = ""
+	var/lifetime_timer
 	universal_speak = 1
 
+/mob/living/automatedannouncer/New()
+	lifetime_timer = addtimer(src, "autocleanup", SecondsToTicks(10))
+	..()
+
+/mob/living/automatedannouncer/Destroy()
+	if(lifetime_timer)
+		deltimer(lifetime_timer)
+	..()
+
+/mob/living/automatedannouncer/proc/autocleanup()
+	log_debug("An announcer somehow managed to outlive the radio! Deleting! Area: [get_area(src)], Loc: [src.loc.loc]"

```

Comment:
Somehow I get the idea this is just another possible runtime again. Goodbye, loc term